plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -202,6 +215,82 @@ private static boolean isAggregate(ExpressionContext expressionContext)
         return expressionContext.getType() == ExpressionContext.Type.FUNCTION && expressionContext.getFunction().getType() == FunctionContext.Type.AGGREGATION;
     }
 
+    private static boolean hasAggregate(ExpressionContext expressionContext)
+    {
+        switch (expressionContext.getType()) {
+            case IDENTIFIER:
+            case LITERAL:
+                return false;
+            case FUNCTION:
+                if (isAggregate(expressionContext)) {
+                    return true;
+                }
+                for (ExpressionContext argument : expressionContext.getFunction().getArguments()) {
+                    if (hasAggregate(argument)) {
+                        return true;
+                    }
+                }
+                return false;
+        }
+        throw new PinotException(PINOT_EXCEPTION, Optional.empty(), format("Unsupported expression type '%s'", expressionContext.getType()));
+    }
+
+    private static Map<String, PinotColumnNameAndTrinoType> getAggregateTypes(SchemaTableName schemaTableName, QueryContext queryContext, Map<String, ColumnHandle> columnHandles)
+    {
+        // A mapping from pinot expression to the returned pinot column name and trino type
+        // Note: the column name is set by the PostAggregationHandler
+        List<ExpressionContext> aggregateColumnExpressions = queryContext.getSelectExpressions().stream()
+                .filter(DynamicTableBuilder::hasAggregate)
+                .collect(toImmutableList());
+        queryContext = new QueryContext.Builder()
+                .setAliasList(queryContext.getAliasList())
+                .setSelectExpressions(aggregateColumnExpressions)
+                .build();
+        DataSchema preAggregationSchema = getPreAggregationDataSchema(queryContext);
+        PostAggregationHandler postAggregationHandler = new PostAggregationHandler(queryContext, preAggregationSchema);
+        DataSchema postAggregtionSchema = postAggregationHandler.getResultDataSchema();
+        ImmutableMap.Builder<String, PinotColumnNameAndTrinoType> aggregationTypesBuilder = ImmutableMap.builder();
+        for (int index = 0; index < postAggregtionSchema.size(); index++) {
+            aggregationTypesBuilder.put(
+                    // ExpressionContext#toString performs quoting of literals
+                    // Quoting of identifiers is not done and this is required to match
+                    // the corresponding column name in the ResultTable returned from Pinot
```

Comment:
Is my suggestion true?

```suggestion
                    // Quoting of identifiers is not done to match the corresponding column name in the ResultTable returned from Pinot. Quoting will be done by `DynamicTablePqlExtractor`.
```